src/sentry/features/manager.py

```diff
@@ -2,26 +2,24 @@
 
 __all__ = ["FeatureManager"]
 
-import itertools
-
+from collections import defaultdict
 from django.conf import settings
 
-from sentry.utils.safe import safe_execute
-
 from .base import Feature
 from .exceptions import FeatureNotRegistered
 
 
 class FeatureManager(object):
     def __init__(self):
-        self._registry = {}
+        self._feature_registry = {}
+        self._handler_registry = defaultdict(list)
```

Comment:
Oops, just realized this needs to be `list` and not `set` if we're going to apply the handlers in a consistent order as the `FeatureManager.has` docstring promises.